lib/ecto/adapters/sql.ex

```diff
@@ -596,6 +602,23 @@ defmodule Ecto.Adapters.SQL do
     end
   end
 
+  ## Migrations
+
+  @doc false
+  def lock_for_migrations(repo, opts, fun) do
+    alias Ecto.Migration.SchemaMigration
+    import Ecto.Query, only: [from: 2]
+
+    transaction(repo, opts, fn ->
+      from(p in {SchemaMigration.get_source(repo), SchemaMigration}, select: p.version)
```

Comment:
I don't like this dependency on schema migrations, it feels a bit backwards. Maybe we can pass the initial query as argument to `lock_for_migrations`? Also, I think we don't even need the `SchemaMigration` name, we can probably write it as:

    from(p in SchemaMigration.get_source(repo), select: type(p.version, :integer))

Other than that, the callback looks great! :+1:

Reply:
So, I could pass the query, but not every adapter would use the same query. If I instead passed the table name, is there a form of calling `execute/6`, that would work? The actually query for the adapters is static aside from the table name.